e2e/test/scenarios/dashboard-filters/dashboard-filters-auto-apply.cy.spec.js

```diff
@@ -113,22 +120,25 @@ describe("scenarios > dashboards > filters > auto apply", () => {
       cy.wait("@updateDashboard");
       cy.findByLabelText("Auto-apply filters").should("be.checked");
     });
-    filterWidget().within(() => {
-      cy.findByText("2 selections").should("be.visible");
-      cy.get("@cardQuery.all").should("have.length", 3);
-    });
+    filterWidget().findByText("Widget").should("be.visible");
+    getDashboardCard().findByText("Rows 1-4 of 54").should("be.visible");
+    cy.get("@cardQuery.all").should("have.length", 4);
```

Comment:
Previously, the new filter `["Widget"]` would be discarded and it would revert back to `["Widget", "Gadget"]`.

This test is changed because we'll now apply unapplied filters when turning on the dashboard auto-apply filters.

`@cardQuery.all` API is also called again once we turn on auto-apply filters on this dashboard since there were un applied filters.

Another way to think about this behavior is like we manually hit the apply button before turning on auto-apply filters.